.github/workflows/build-test.yaml

```diff
@@ -0,0 +1,50 @@
+name: Build Test
+
+on:
+  push:
+    branches: [main]
+    paths-ignore:
+      - README.md
+  pull_request:
+    branches: [main]
+    paths-ignore:
+      - README.md
+
+jobs:
+  Build-Test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+          token: ${{ secrets.HEADLESS_USER_PAT }}
+
+      - uses: actions/setup-go@v2
+        with:
+          go-version: "1.18"
+      - name: Setup blosc library
+        run: |
+          sudo apt-get install -y libblosc-dev
+      - name: Run Tests
+        run: |
+          go test ./... -coverprofile=coverage.out -covermode=atomic
+      - name: Upload Coverage
+        uses: codecov/codecov-action@v2
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          fail_ci_if_error: true
+          files: ./coverage.out
+
+      - name: Calc Semver
+        id: semver
+        uses: paulhatch/semantic-version@v4.0.2
+        with:
+          tag_prefix: "v"
+          major_pattern: "(MAJOR)"
+          minor_pattern: "(MINOR)"
+          format: "${major}.${minor}.${patch}"
+          short_tags: false
+      - name: Print Version tag
+        run: |
+          echo "Next Version to be Released: ${{ steps.semver.outputs.version_tag }}"
```

Comment:
Just thinking it might be helpful for this to post as a comment instead, but just a nice to have.

https://github.com/marketplace/actions/add-pr-comment

Reply:
yep, easy enough to add in.